### PR TITLE
All Furnitures with linksToNeighbour are now properly deconstructed and their Jobs are canceled

### PR DESCRIPTION
**Pull request by BeefEX**
Added code that updates all tiles with same furniture around
deconstructed furniture.

### CODE REVIEW

**Comment by BeefEX** [comment]
Sorry for messy comitts, i was messing with github :smile:

**Comment by lameox** [comment]
You should probably squash your last 3 commits into one, so they dont pollute the history. Alternatively you can revert your branch by two commits which serves the same purpose.

**Comment by BeefEX** [comment]
I dont know what is going on, i have huge mess in my github. How can i squash them?

**Comment by quill18** [comment]
@lameox I have forced squashing for the actual merge into the main project -- so we shouldn't have to worry about things getting too messy.

**Comment by BeefEX** [comment]
@quill18 thats nice for the noobies to git as i am :D thx

**Comment by quill18** [comment]
@BeefEX Heya!  Can you add in a few comments please?  Just at the start of each major block should be sufficient.

**Comment by BeefEX** [comment]
@quill18 why not wait a minute

**Comment by BeefEX** [comment]
@quill18 done, and fixed the code too, but my commits are very messy :/

File: Assets/Scripts/Models/Tile.cs [hunk 1]
```diff
@@ -124,25 +124,51 @@ public void UnregisterTileTypeChangedCallback(Action<Tile> callback)
...
+        f.CancelJobs (); // Canceles all jobs by this furniture.
+
         for (int x_off = X; x_off < (X + f.Width); x_off++)
         {
             for (int y_off = Y; y_off < (Y + f.Height); y_off++)
             {
 
                 Tile t = World.current.GetTileAt(x_off, y_off);
+                if (t.furniture.linksToNeighbour)
+                    affectedTiles.Add (t);
```
Comment (lameox) [inline Assets/Scripts/Models/Tile.cs]: I think you can discard the affectedTiles list  and just call t.cbOnChanged(t.furniture) right here

Comment (BeefEX) [inline Assets/Scripts/Models/Tile.cs]: there is another way let me try it

**Comment by BeefEX** [comment]
i have some problems with commiting right now, but here is the final code:
`for (int x_off = X; x_off < (X + f.Width); x_off++)
        {
            for (int y_off = Y; y_off < (Y + f.Height); y_off++)
            {

                Tile t = World.current.GetTileAt(x_off, y_off);
				if (t.furniture.linksToNeighbour)
				{
					//This will update all neighbour tiles.
					foreach (int[] offset in offsets) // Using hardcoded offsets. Is there better way?
					{ 
						Tile tmpTile = World.current.GetTileAt (t.X + offset[0], t.Y + offset[1]);
						if (tmpTile.furniture != null && tmpTile.furniture.objectType == f.objectType)
						{
							tmpTile.furniture.cbOnChanged (tmpTile.furniture);
						}
					}
				}
				t.furniture = null;
            }
        }`

**Comment by rfmsimoes** [comment]
Hello,

@BeefEX  in my opion I think you don't need (and probably it is not the best place) execute the function CancelJobs() inside the UnplaceFurniture function in Tile.cs but what you need to do is go to Furniture.cs and in the fucntion Deconstruct() before call the tile.UnplaceFurniture() call the CancelJobs fuctions.  

Because when you deconstruct furnitures, after the click or dragging the mouseController will call the tile.furniture.Deconstruct()

=)

**Comment by BeefEX** [comment]
@rfmsimoes will update it when i get home thx for pointing that out!

**Comment by BeefEX** [comment]
PR #92 is better. Closing this one.

**Pull request closed without merging**